azure-pipelines/main-pipeline.yml

```diff
@@ -0,0 +1,228 @@
+      - task: SonarCloudPublish@1
+        inputs:
+          pollingTimeoutSec: '300'
+          
+      - task: PublishPipelineArtifact@1
+        inputs:
+          targetPath: '$(Build.Repository.LocalPath)/base/target/base-1.jar'
+          artifact: 'base'
+          publishLocation: 'pipeline'
+
+      - script:  echo 'Built base'
+
+# Here we're looping through the objects in our parameters. This serves as a way to name jobs (which require unique names)
+# that are created as part of the loop. The job checks if a service is to be built, or skips the job.
+
+  - ${{ each service in parameters.serviceConditions }}:
+    - job: ${{ service.jobName }}
+      pool:
+        name: java-16
+      dependsOn: BaseBuild
+
+# Setting Variables that contain values for the conditional logic below.
+
+      variables:
+         accounts: $[ dependencies.CheckChanges.outputs['ProduceVar.accounts'] ]
+         base: $[ dependencies.CheckChanges.outputs['ProduceVar.base'] ]
+         commerce: $[ dependencies.CheckChanges.outputs['ProduceVar.commerce'] ]
+         config: $[ dependencies.CheckChanges.outputs['ProduceVar.config'] ]
+         eureka-server: $[ dependencies.CheckChanges.outputs['ProduceVar.eureka-server'] ]
+         inventory: $[ dependencies.CheckChanges.outputs['ProduceVar.inventory'] ]
+         Spring-Cloud-Gateway: $[ dependencies.CheckChanges.outputs['ProduceVar.Spring-Cloud-Gateway'] ]
+
+# Jobs created dynamically will only run if they were marked True, previously.
+
+      condition: eq(variables['${{ service.name }}'], 'True')
+      steps:
+      - script: echo '${{ service.name }} has been updated - $(${{ service.name }})'
+        displayName:  'Checking ${{ service.name }}'
+
+# Download Base Artifact to install wich dependant services.
+
+      - task: DownloadPipelineArtifact@2
+        inputs:
+          buildType: 'current'
+          artifactName: 'base'
+          itemPattern: '**/*.jar'
+          targetPath: '$(Build.Repository.LocalPath)'
+
+
+      - task: SonarCloudPrepare@1
+        inputs:
+          SonarCloud: 'SonarCloud'
+          organization: '$(org)'
+          scannerMode: 'CLI'
+          configMode: 'manual'
+          cliProjectKey: '$(projectKey)'
+          cliProjectName: '$(projectName)'
+          cliSources: '.'
+          extraProperties: 'sonar.java.binaries=**/target/classes'
+
+
+# Package Individual Service to create artifacts from.  
+
+      - script: |      
+
+          if [ ${{ service.requiresBase }} = true ]
+          then
+            mvn install:install-file
+            -Dfile="$(Build.Repository.LocalPath)/base-1.jar"
+            -DgroupId="com.revature"
+            -DartifactId="base"
+            -Dversion="1"
+            -Dpackaging=jar
+            -DgeneratePom=true;
+            
+            ls -a ~/.m2/repository/com/revature/base/1/
+          fi
+             mvn package -f './${{ service.name }}/pom.xml'
+          
+        name: BuildScript
```

Comment:
without `set -e` or some other mechanism, an early nonzero exit code will be ignored, and only the last command's exit code will be noticed. if you're confident that will work appropriately then ok

Reply:
I used -e for testing purposes, and I'm currently confident with  or without using it, but you've raised a good point. Is this something standard to keep in a pipeline?

Reply:
for some people yes, for others no.

Reply:
Added this.

Reply:
just in case we're not on the same page, the `-e` option to `mvn` is completely different from what i was talking about

Reply:
I was on the wrong page. Definitely misinterpreted that, and I understand now.